src/python/pants/build_graph/target.py

```diff
@@ -632,6 +638,40 @@ def traversable_dependency_specs(self):
     """
     return []
 
+  @classmethod
+  def compute_dependency_specs(cls, kwargs=None, payload=None):
+    """Given either pre-Target.__init__() kwargs or a post-Target.__init__() payload, compute the
+    full set of dependency specs in the same vein as the prior `traversable_dependency_specs`.
+
+    N.B. This is a temporary bridge to span the gap between v2 "Fields" products vs v1 `BuildGraph`
+    `Target` object representations. See:
+
+      https://github.com/pantsbuild/pants/issues/3560
+      https://github.com/pantsbuild/pants/issues/3561
+
+    :param dict kwargs: The pre-Target.__init__() kwargs dict.
+    :param Payload payload: The post-Target.__init__() Payload object.
+    :yields: Spec strings representing dependencies of this target.
+    """
+    assert not all((kwargs is None, payload is None)), 'must provide either kwargs or payload'
+    assert not all((kwargs is not None, payload is not None)), 'may not provide both kwargs and payload'
+    assert not (kwargs and not isinstance(kwargs, dict)), (
+      'expected a `dict` object for kwargs, instead found a {}'.format(type(kwargs))
+    )
+    assert not (payload and not isinstance(payload, Payload)), (
+      'expected a `Payload` object for payload, instead found a {}'.format(type(payload))
+    )
+    # N.B. This pattern turns this method into a non-yielding generator, which is helpful for subclassing.
+    return
+    yield
+
+  @classmethod
+  def apply_injectables(cls, build_graph):
+    """Given a BuildGraph, apply all of this targets subsystems injectables."""
+    for subsystem in cls.subsystems():
```

Comment:
At least in `v2`, we always inject as a batch though (even if the root was a `SingleAddress`, we're going to have its transitive deps)... I think minimizing re-injection will be important, unless I'm missing something.